.github/workflows/ci.yml

```diff
@@ -1,64 +1,64 @@
 name: CI
 
 on:
-    push:
-      branches:
-        - master
-    pull_request:
+  push:
+    branches:
+      - master
+  pull_request:
 
 concurrency:
-    group: ${{ github.workflow }}-${{ github.ref }}
-    cancel-in-progress: true
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
 
 jobs:
-    python_test:
-      runs-on: ubuntu-latest
-      steps:
-        - uses: actions/checkout@v4
-        - name: Install Python
-          uses: actions/setup-python@v4
-          # see details (matrix, python-version, python-version-file, etc.)
-          # https://github.com/actions/setup-python
-          with:
-            python-version: '3.11'
-        - name: Install poetry
-          uses: abatilo/actions-poetry@v2
-        - name: Setup a local virtual environment (if no poetry.toml file)
-          run: |
-            poetry config virtualenvs.create true --local
-            poetry config virtualenvs.in-project true --local
-        - uses: actions/cache@v3
-          name: Define a cache for the virtual environment based on the dependencies lock file
-          with:
-            path: ./.venv
-            key: venv-${{ hashFiles('poetry.lock') }}
-        - name: Install the project dependencies
-          run: poetry install --with=dev
-        - name: pytest
-          run: poetry run pytest -v
-          env:
-            OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
-        - name: check openapi client up to date
-          run: |
-            poetry run poe extract-openapi
-            poetry run poe update-client
-            # Record intent to add any new files in client
-            git add -N dewy-client
-            # Diff, and report any changes (including any new files in dewy-client)
-            git diff --exit-code
+  python_test:
+    runs-on: ubuntu-latest
```

Comment:
Is there not a way to run in an environment with Python already available?

Reply:
No -- these are OSes (eg., mac/ubuntu/etc.) not docker images.

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idruns-on